### PR TITLE
#383 Ayah Play Audio + Redux Refactoring (WIP)

### PR DESCRIPTION
**Pull request by sabeurthabti**
Hi, 

This will be a slightly *fat* PR. I am not a fan of the decorators, I will make a suggested change and let me know if you agree. I find wrapping the component in a normal function is much easier for new contributors who probably don't use bleed-edge of ES. Also they have not made it into the official spec. Thoughts?

- [x] Clicking the Play link should play the current ayah.
- [x] Refactoring out decorators
- [x] Hide play in search results

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32770

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32771

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32772

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32773

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32774

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32775

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32776

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32777

File: src/components/Ayah/index.js [hunk 1]
```diff
@@ -44,10 +45,14 @@ export default class Ayah extends Component {
...
-      open: false
-    });
+  handlePlay(ayah) {
+
+    const {stop, setAyah, start} = this.props.actions;
+
+    stop();
+    setAyah(ayah);
+    start();
+
```
Comment (mmahalwy) [inline src/components/Ayah/index.js]: ✂️

File: src/components/Ayah/index.js [hunk 2]
```diff
@@ -44,10 +45,14 @@ export default class Ayah extends Component {
     return conditions.some(condition => condition);
   }
 
-  handlePlay() {
-    this.setState({
-      open: false
-    });
+  handlePlay(ayah) {
+
```
Comment (mmahalwy) [inline src/components/Ayah/index.js]: ✂️

File: src/containers/App/index.js [hunk 3]
```diff
@@ -18,12 +18,7 @@ import FontStyles from 'components/FontStyles';
 const styles = require('./style.scss');
 
 @metrics(metricsConfig)
```
Comment (mmahalwy) [inline src/containers/App/index.js]: you forgot this one!

File: src/containers/Home/index.js [hunk 4]
```diff
@@ -189,4 +177,16 @@ class Home extends Component {
...
+  promise({ store: { getState, dispatch } }) {
+    if (!isAllLoaded(getState())) {
+      return dispatch(loadAll());
+    }
+
+    return true;
+  }
+}])(Home);
+
+
```
Comment (mmahalwy) [inline src/containers/Home/index.js]: ✂️ extra line

File: src/containers/Surah/index.js [hunk 5]
```diff
@@ -505,3 +459,39 @@ export default class Surah extends Component {
...
+    isLoading: state.ayahs.loading,
+    isLoaded: state.ayahs.loaded,
+    lines: state.lines.lines,
+    options: state.options
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: {
```
Comment (mmahalwy) [inline src/containers/Surah/index.js]: well done! Great idea to do it this way

File: src/redux/modules/audioplayer.js [hunk 6]
```diff
@@ -171,6 +172,27 @@ export default function reducer(state = initialState, action = {}) {
         currentTime: 0
       };
     }
+
+    case SET_AYAH: {
+
```
Comment (mmahalwy) [inline src/redux/modules/audioplayer.js]: ✂️

**Comment by mmahalwy** [comment]
@sabeurthabti this is awesome mashallah! Thanks for this

re decorators: i do really like how you separated the `mapStateToProps` methods, should definitely do that myself too. As for decorators, I do see your point about them being bleeding edge. One thing that separating them will benefit us is testability, more specifically see: http://redux.js.org/docs/recipes/WritingTests.html#connected-components

I do see your point but I really do like decorators, so I am torn. I think we should make a call and go with it. What do you think?

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32778

**Comment by sabeurthabti** [comment]
@mmahalwy let's discuss it on slack. I am happy with either.

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32779

**Pull request merged**